Add database drawing and schema

w286432531 opened the pull request:
Finish database drawing.
Add database models.

w286432531 committed:
Add database drawing and schema

linmao0991 commented on prisma/schema.prisma:
```diff
@@ -11,18 +11,166 @@ datasource db {
+model BusinessHour {
+  id    Int    @id @default(autoincrement())
+  day   String
+  hours String
```
We might want to change the types or add fields to this table. 7 days a week with possibility varying times each day. Maybe consider changing hours to `json` we can place a json object with each day and hours.

w286432531 replied:
I changed datatype to time

linmao0991 commented on prisma/schema.prisma:
```diff
@@ -11,18 +11,166 @@ datasource db {
+model UserType {
+  id              Int     @id @default(autoincrement())
+  name            String
+  alternativeName String?
```
What is the purpose of user type? Like roles? Customer, Owner ...etc? If so what is `name` and `alternativeName` ?

w286432531 replied:
I changed model name to role

linmao0991 commented on prisma/schema.prisma:
```diff
@@ -11,18 +11,166 @@ datasource db {
+  itemId       Int
+  option       Option        @relation(fields: [optionId], references: [id])
+  optionId     Int
+  price        Decimal       @db.Decimal(12, 2)
```
Although decimals are fine, consider using just an `Int` here and then you can represent prices in whole numbers. Like $1.00 would be stored as `100`. Then on the front end you can convert these with 2 decimal places when displaying them.

Why I suggest doing this way is because any type of arithmetic done to these will result in rounding errors . Here is a good read on why https://frontstuff.io/how-to-handle-monetary-values-in-javascript.

linmao0991 reviewed (changes requested):
Over all it's a good start. There are a few things I would look over that commented on.

I might also suggest putting some comments in Prisma on the uses for some of these tables. Once it's refined you can move those comments in some kind of documentation for your schema.

w286432531 committed:
Modify schema file and drawing

Changed all decimal to int
Add comments
Replaced add on tables and relationships with ingredients.

linmao0991 reviewed (approved):
Looks good.

**Future stuff**
@w286432531 I see you are accommodating for different languages ( English, Chinese, ...etc)? If that is the case it might be good to think about a different way to approach this rather than a field for the alternate name. Maybe a `itemName`, `ingredientName`, ...etc tables and have  a 1-to-1 relationship with item and ingredients. The name tables could contain fields with each language and names of item or ingredient in each language field.

With this approach ( or something like it ) you can add more than 1 translation of the name and ingredient.
